pkg/columns/formatter/json/json_test.go

```diff
@@ -141,3 +142,47 @@ func TestDynamicFields(t *testing.T) {
 	formatter := NewFormatter[empty](cols.GetColumnMap())
 	assert.Equal(t, `{"str": "foobar", "int32": 1234567890, "bool": true}`, formatter.FormatEntry(&empty{}))
 }
+
+func TestJSONFormatter(t *testing.T) {
+	type testStruct struct{}
+
+	cols := columns.MustCreateColumns[testStruct]()
+
+	assert.NoError(t, cols.AddColumn(columns.Attributes{
+		Name: "parent.child1.grandchild1",
+	}, func(t *testStruct) string { return "parent.child1.grandchild1_text" }))
+	assert.NoError(t, cols.AddColumn(columns.Attributes{
+		Name: "parent.child1.grandchild2",
+	}, func(t *testStruct) string { return "parent.child1.grandchild2_text" }))
+	assert.NoError(t, cols.AddColumn(columns.Attributes{
+		Name: "parent.child2.grandchild1",
+	}, func(t *testStruct) string { return "parent.child2.grandchild1_text" }))
+	assert.NoError(t, cols.AddColumn(columns.Attributes{
+		Name: "parent.child3",
+	}, func(t *testStruct) string { return "parent.child3_text" }))
+
+	assert.NoError(t, cols.AddColumn(columns.Attributes{
+		Name: "parent.child1",
+	}, func(t *testStruct) string { return "This should be skipped/filtered" }))
+
+	expected := `{
+		"parent": {
+			"child1": {
+				"grandchild1": "parent.child1.grandchild1_text",
+				"grandchild2": "parent.child1.grandchild2_text"
+			},
+			"child2": {
+				"grandchild1": "parent.child2.grandchild1_text"
+			},
+			"child3": "parent.child3_text"
+		}
+	}`
+
+	formatter := NewFormatter[testStruct](cols.ColumnMap)
+	actual := formatter.FormatEntry(&testStruct{})
+	assert.JSONEq(t, expected, actual)
+
+	prettyFormatter := NewFormatter[testStruct](cols.ColumnMap, WithPrettyPrint())
+	actual = prettyFormatter.FormatEntry(&testStruct{})
+	assert.JSONEq(t, expected, actual)
```

Comment:
Shouldn't this check that the printed json has the same format? I mean, this is for testing pretty printed json, so I think it should compare the string itself and not only the json, otherwise the test will pass even if it's not pretty printed.

Reply:
In theory yes - it should have the same structure.
The problem is with the order of the children. This is (currently) not ordered lexicographical. Therefore I don't know which children comes first and which comes last. And if it is different between runs or host systems